File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter04/JAVA/Distance.java

/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 04.-Flujo de control I: Estructuras Selectivas
    Problema 23.-Si distancia es mayor que 20 y menos que 35, leer un valor para tiempo.

    Ejercicios
    4.1 (e) Pagina 155
*/
import java.util.Scanner;

public class Distance{

    public static void main(String[] args){
        Scanner in=new Scanner(System.in);
        int distance=0, time=0;
    
        System.out.print("programa Si distancia es mayor que 20 y menos que 35, leer un valor para tiempo: ");
    
        System.out.print("\n\nIngresa la distancia: ");
        distance=in.nextInt();
    
        if(distance>20 && distance<35){
            System.out.print("ingresa el tiempo: ");
            time=in.nextInt();
        }
    
        System.out.print("\nLos valores son:\nDistancia = "+distance+"\nTiempo = "+time+"\n\n");
    
    }
}

File: FundamentosDeLaProgramacionLuisJoyanes4taEdicion/chapter04/JAVASCRIPT/js/distance.js

/*Libro:Fundamentos de Programación, Algoritmos, estructura de datos y objetos 4ta edición. 
    Capitulo 04.-Flujo de control I: Estructuras Selectivas
    Problema 23.-Si distancia es mayor que 20 y menos que 35, leer un valor para tiempo.

    Ejercicios
    4.1 (e) Pagina 155
*/
function calculate(){
    let distance=0, time=0;
    
    var varSpan=document.createElement("span");
        varSpan.setAttribute("id", "idSpan");
        
    if(document.getElementById("idBtnCalculate").value=="Calcular"){                     
        

            distance=parseInt(document.getElementById("idDistance").value);

            if(distance>20 && distance<35){
                document.getElementById("idTime").disabled=false;
            }else{
                alert("No cumple la condición ");
            }
        document.getElementById("idSection").appendChild(varSpan);    
        document.getElementById("idBtnCalculate").value="Limpiar";
    }else{
        document.getElementById("idDistance").value="";
        document.getElementById("idTime").value="";
        document.getElementById("idTime").disabled=true;
        document.getElementById("idSpan").remove();


        document.getElementById("idBtnCalculate").value="Calcular";
    }
}